agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -178,491 +91,303 @@ public CallReplacementTransformer() {}
+      if (transformClass(cg)) {
+        JavaClass javaClass = cg.getJavaClass();
+        if (MapCallsAgent.debug) {
+          Path filepath = MapCallsAgent.debugPath.resolve(javaClass.getClassName() + ".class");
+          javaClass.dump(filepath.toFile());
+        }
+        debug_transform.log("transform: EXIT class %s transformed", className);
+        return javaClass.getBytes();
       } else {
-        debug_transform.log("not including class %s (filtered out)", className);
+        debug_transform.log("transform: EXIT class %s not transformed", className);
         return null;
       }
-
     } catch (Throwable e) {
-      out.format("Unexpected error %s in transform", e);
+      debug_transform.log(
+          "transform: EXIT transform of %s resulted in exception %s%n", className, e);
+      System.out.format(
+          "Unexpected exception %s (%s) in class transform of %s%n", e, e.getCause(), className);
       e.printStackTrace();
-      return (null);
+      return null;
     }
   }
 
   /**
-   * Processes each method in cg replacing any specified calls with static user calls.
+   * Indicates whether the class is a non-AWT/Swing boot-loaded class.
+   *
+   * <p>This check is for performance, since attempting to transform all of {@code rt.jar} is
+   * unnecessary. However, we want to transform AWT/Swing classes if they are loaded.
    *
-   * @param fullClassName must be packageName.className
+   * <p>Actually checks whether the class is loaded by the bootstrap loader or by the first
+   * classloader. The first classloader will either be a user-provided boot loader, or the extension
+   * class loader. Since the predicate is mostly for performance, we don't make the extra check to
+   * determine if the user has given a boot loader.
+   *
+   * @param loader the class loader for the class
+   * @param fullClassName the fully-qualified name of the class
+   * @return true if the named class is boot-loaded and not in {@code java.awt.} or {@code
+   *     javax.swing.}, false otherwise
    */
-  private boolean map_calls(
-      ClassGen cg,
-      String fullClassName,
-      ClassLoader loader,
-      Map<MethodDef, ReplacementClass> current_method_map) {
+  private boolean isNonGUIBootClass(ClassLoader loader, String fullClassName) {
+    return (loader == null || loader.getParent() == null)
+        && !(fullClassName.startsWith("java.awt.") || fullClassName.startsWith("javax.swing."));
+  }
+
+  /**
+   * Indicates whether the named class occurs in a package that is excluded. Tests whether one of
+   * the excluded package names is a prefix of the fully-qualified class name.
+   *
+   * @param fullClassName the fully-qualified class name, must be non-null
+   * @return true if any excluded package is a prefix of the class name, false otherwise
+   */
+  private boolean isExcludedClass(ClassLoader loader, String fullClassName) {
+    if (isNonGUIBootClass(loader, fullClassName)) {
+      return false;
```

Comment:
It looks to me like this should be `true`:  non-GUI boot-loaded classes should not be transformed.
Why is it `false`?